dipy/segment/tests/test_clustering.py

```diff
@@ -0,0 +1,575 @@
+
+
+def test_cluster_map_remove_cluster():
+    clusters = ClusterMap()
+
+    cluster1 = Cluster(indices=[1])
+    clusters.add_cluster(cluster1)
+
+    cluster2 = Cluster(indices=[1, 2])
+    clusters.add_cluster(cluster2)
+
+    cluster3 = Cluster(indices=[1, 2, 3])
+    clusters.add_cluster(cluster3)
+
+    assert_equal(len(clusters), 3)
+
+    clusters.remove_cluster(cluster2)
+    assert_equal(len(clusters), 2)
+    assert_array_equal(list(itertools.chain(*clusters)), list(itertools.chain(*[cluster1, cluster3])))
+    assert_equal(clusters[0], cluster1)
+    assert_equal(clusters[1], cluster3)
+
+    clusters.remove_cluster(cluster3)
+    assert_equal(len(clusters), 1)
+    assert_array_equal(list(itertools.chain(*clusters)), list(cluster1))
+    assert_equal(clusters[0], cluster1)
+
+    clusters.remove_cluster(cluster1)
+    assert_equal(len(clusters), 0)
+    assert_array_equal(list(itertools.chain(*clusters)), [])
+
+
+def test_cluster_map_iter():
+    nb_clusters = 11
+
+    # Test without specifying refdata in ClusterMap
+    cluster_map = ClusterMap()
+    clusters = []
+    for i in range(nb_clusters):
+        new_cluster = Cluster(indices=np.random.randint(0, len(data), size=10))
+        cluster_map.add_cluster(new_cluster)
+        clusters.append(new_cluster)
+
+    assert_true(all([c1 is c2 for c1, c2 in zip(cluster_map.clusters, clusters)]))
+    assert_array_equal(cluster_map, clusters)
+    assert_array_equal(cluster_map.clusters, clusters)
+    assert_array_equal(cluster_map, [cluster.indices for cluster in clusters])
+
+    # Set refdata
+    cluster_map.refdata = data
+    assert_array_equal(cluster_map, [[data[i] for i in cluster.indices] for cluster in clusters])
+
+
+def test_cluster_map_getitem():
+    nb_clusters = 11
+    indices = list(range(len(data)))
+    np.random.shuffle(indices)  # None trivial ordering
+    advanced_indices = indices + [0, 1, 2, -1, -2, -3]
+
+    cluster_map = ClusterMap()
+    clusters = []
+    for i in range(nb_clusters):
+        new_cluster = Cluster()
+        cluster_map.add_cluster(new_cluster)
+        clusters.append(new_cluster)
+
+    # Test indexing
+    for i in advanced_indices:
+        assert_equal(cluster_map[i], clusters[i])
+
+    # Test advanced indexing
+    assert_array_equal(cluster_map[advanced_indices], [clusters[i] for i in advanced_indices])
+
+    # Test index out of bound
+    assert_raises(IndexError, cluster_map.__getitem__, len(clusters))
+    assert_raises(IndexError, cluster_map.__getitem__, -len(clusters)-1)
+
+    # Test slicing and negative indexing
+    assert_equal(cluster_map[-1], clusters[-1])
+    assert_array_equal(cluster_map[::2], clusters[::2])
+    assert_arrays_equal(cluster_map[::-1], clusters[::-1])
+    assert_arrays_equal(cluster_map[:-1], clusters[:-1])
+    assert_arrays_equal(cluster_map[1:], clusters[1:])
+
+
+def test_cluster_map_comparison_with_int():
```

Comment:
An `int`. When a ClusterMap object is being compared to an int `N` it returns a boolean array indicating which cluster contains more than `N` elements.